### PR TITLE
fix(selfservice): Login self service flow with TOTP does not pass on return_to URL

### PR DESCRIPTION
**Pull request by sawadashota**
## Related issue(s)

#2172

## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md).
- [x] I have referenced an issue containing the design document if my change
      introduces a new feature.
- [x] I am following the
      [contributing code guidelines](../blob/master/CONTRIBUTING.md#contributing-code).
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security. vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature
      works.
- [ ] I have added or changed [the documentation](docs/docs).

### CODE REVIEW

File: selfservice/flow/login/hook.go [hunk 1]
```diff
@@ -60,10 +60,18 @@ func NewHookExecutor(d executorDependencies) *HookExecutor {
...
+func (e *HookExecutor) requiresAAL2(r *http.Request, s *session.Session, a *Flow) (*session.ErrAALNotSatisfied, bool) {
 	var aalErr *session.ErrAALNotSatisfied
 	err := e.d.SessionManager().DoesSessionSatisfy(r, s, e.d.Config(r.Context()).SessionWhoAmIAAL())
-	return aalErr, errors.As(err, &aalErr)
+	if ok := errors.As(err, &aalErr); !ok {
+		return nil, false
+	}
+
+	if err := aalErr.PassReturnToParameter(a.RequestURL); err != nil {
+		return aalErr, true
```
Comment (kszafran) [inline selfservice/flow/login/hook.go:71]: Wasn't this supposed to be:
```suggestion
		return err, true
```
?

Comment (sawadashota) [reply selfservice/flow/login/hook.go]: Thanks! I fixed.

File: session/manager.go [hunk 2]
```diff
@@ -41,6 +42,28 @@ func (e *ErrAALNotSatisfied) EnhanceJSONError() interface{} {
 	return e
 }
 
+func (e *ErrAALNotSatisfied) PassReturnToParameter(requestURL string) error {
```
Comment (aeneasr) [inline session/manager.go:45]: Could you please add a dedicated, small test for this? :)

**Review by aeneasr — CHANGES_REQUESTED** [review]
This looks good to go! Just one little test :)

File: session/manager.go (hunk 2 above)
Comment (sawadashota) [reply session/manager.go:45]: OK! I've added.

**Review by aeneasr — APPROVED** [review]
Thank you for your great contributions, as always! /cc @vinckr not sure if we've sent out some swag to @sawadashota already? :)

**Comment by vinckr** [comment]
Hello @sawadashota 

As Aeneas mentioned we should have sent you some swag ages ago! 
You are one of the earliest and steadiest contributors to Ory, many many thanks for your work 🙇 
Your contributions are helping secure millions of identities around the globe 🌏. 

As a small token of appreciation we would love to send you a gift package. 
[Please drop me an email and I will forward you the form to claim your Ory swag!](mailto:vincent@ory.sh)

**Pull request merged**